src/SqlBindingUtilities.cs

```diff
@@ -136,11 +135,30 @@ public static SqlCommand BuildCommand(SqlAttribute attribute, SqlConnection conn
         /// <summary>
         /// Returns a dictionary where each key is a column name and each value is the SQL row's value for that column
         /// </summary>
-        /// <param name="reader">Used to determine the columns of the table as well as the next SQL row to process</param>
+        /// <param name="reader">
+        /// Used to determine the columns of the table as well as the next SQL row to process
+        /// </param>
+        /// <param name="cols">
+        /// Filled with the columns of the table if empty, otherwise assumed to be populated
+        /// with their names already (used for cacheing so we don't retrieve the column names every time)
+        /// </param>
         /// <returns>The built dictionary</returns>
-        public static IReadOnlyDictionary<string, string> BuildDictionaryFromSqlRow(SqlDataReader reader)
+        public static Dictionary<string, string> BuildDictionaryFromSqlRow(SqlDataReader reader, List<string> cols)
```

Comment:
I think we should leave this change as it is also affecting trigger-binding logic.